texar/torch/modules/classifiers/bert_classifier.py

```diff
@@ -241,11 +241,11 @@ def forward(self,  # type: ignore
                   ``[batch_size, max_time, num_classes]`` and ``pred`` is of
                   shape ``[batch_size, max_time]``.
         """
-        if self.__class__.__name__.startswith('BERT'):
+        if self._MODEL_NAME == 'BERT':
             enc_outputs, pooled_output = self._encoder(inputs,
                                                        sequence_length,
                                                        segment_ids)
-        elif self.__class__.__name__.startswith('RoBERTa'):
+        elif self._MODEL_NAME == 'RoBERTa':
             enc_outputs, pooled_output = self._encoder(inputs,
                                                        sequence_length)
```

Comment:
And I don't think we need this check at all. Don't we ignore `segment_ids` in `RoBERTaEncoder`?

Reply:
`RoBERTaEncoder` only accepts two arguments (`inputs` and `sequence_length`). Do you mean we add `segment_ids` back in `RoBERTaEncoder` and set it always to be `None`?

Reply:
Since the classifier already does this, I think we can add it back.

Reply:
We could check that `segment_ids` is `None` and give a warning otherwise.